### PR TITLE
[ffmpeg] Remove the download address of gcc under windows arm

### LINKED ISSUE #26538: [FFmpeg] build failure
**Host Environment**
- OS: Windows
- Compiler: gcc

**To Reproduce**
`./vcpkg install ffmpeg:x64-windows`

**Failure logs**
```
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'manifests' = off
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Downloading https://repo.msys2.org/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz
[DEBUG] Downloading https://www2.futureware.at/~nickoe/msys2-mirror/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz
[DEBUG] Downloading https://mirror.yandex.ru/mirrors/msys2/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz
[DEBUG] Downloading https://mirrors.tuna.tsinghua.edu.cn/msys2/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz
[DEBUG] Downloading https://mirrors.ustc.edu.cn/msys2/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz
[DEBUG] Downloading https://mirror.bit.edu.cn/msys2/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz
[DEBUG] Download failed -- retrying after 1000 ms.
[DEBUG] Download failed -- retrying after 2000 ms.
[DEBUG] Download failed -- retrying after 4000 ms.
[DEBUG] Downloading https://mirror.selfnet.de/msys2/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz
[DEBUG] Downloading https://mirrors.sjtug.sjtu.edu.cn/msys2/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz
Error: Failed to download from mirror set:
https://repo.msys2.org/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz: failed: status code 404
https://www2.futureware.at/~nickoe/msys2-mirror/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz: failed: status code 404
https://mirror.yandex.ru/mirrors/msys2/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz: failed: status code 404
https://mirrors.tuna.tsinghua.edu.cn/msys2/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz: failed: status code 404
https://mirrors.ustc.edu.cn/msys2/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz: failed: status code 404
https://mirror.bit.edu.cn/msys2/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz: WinHttpSendRequest() failed: 12007
https://mirror.bit.edu.cn/msys2/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz: WinHttpSendRequest() failed: 12007
https://mirror.bit.edu.cn/msys2/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz: WinHttpSendRequest() failed: 12007
https://mirror.bit.edu.cn/msys2/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz: WinHttpSendRequest() failed: 12007
https://mirror.selfnet.de/msys2/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz: failed: status code 404
https://mirrors.sjtug.sjtu.edu.cn/msys2/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz: failed: status code 404
```

### PR DESCRIPTION
**Pull request by Cheney-W**
**Describe the pull request**

- #### What does your PR fix?
  Fixes https://github.com/microsoft/vcpkg/issues/26538

  Above issue is due to the old version of gcc-9.3.0-1-x86_64 has been removed from https://repo.msys2.org/msys/x86_64.
  After removing the download address of gcc, this port can still be successfully installed on the two triplets arm64-windows and arm-uwp, so remove gcc download.

   Feature test passed with arm-uwp and arm64-windows. (This issue only occur with arm triplet)

### CODE REVIEW

**Comment by Neumann-A** [comment]
I disagree with the assesment that it is necessary to download gcc to run cpp! Run the preprocessor with the correct cl options!

**Comment by Cheney-W** [comment]
@Neumann-A 
Sorry, I don't quite understand what to do, can you give some specific guidance.
I have tested it, and `ffmpeg:arm-uwp` can still be installed successfully without downloading gcc.

File: ports/ffmpeg/portfile.cmake [hunk 1]
```diff
@@ -69,8 +69,8 @@ if(VCPKG_TARGET_IS_WINDOWS)
         vcpkg_acquire_msys(MSYS_ROOT
             DIRECT_PACKAGES
                 # Required for "cpp.exe" preprocessor
```
Comment (Neumann-A) [inline ports/ffmpeg/portfile.cmake:71]: This states that cpp.exe is required which is why gcc is required. I don't agree with that assesment. If it works without downloading gcc at all that is also fine by me

File: ports/ffmpeg/portfile.cmake [hunk 2]
```diff
@@ -70,8 +70,6 @@ if(VCPKG_TARGET_IS_WINDOWS)
...
-                "https://repo.msys2.org/msys/x86_64/gcc-9.3.0-1-x86_64.pkg.tar.xz"
-                76af0192a092278e6b26814b2d92815a2c519902a3fec056b057faec19623b1770ac928a59a39402db23cfc23b0d7601b7f88b367b27269361748c69d08654b2
                 "https://repo.msys2.org/msys/x86_64/isl-0.22.1-1-x86_64.pkg.tar.xz"
                 f4db50d00bad0fa0abc6b9ad965b0262d936d437a9faa35308fa79a7ee500a474178120e487b2db2259caf51524320f619e18d92acf4f0b970b5cbe5cc0f63a2
                 "https://repo.msys2.org/msys/x86_64/zlib-1.2.11-1-x86_64.pkg.tar.xz"
```
Comment (Neumann-A) [inline ports/ffmpeg/portfile.cmake:75]: is the rest here required? if yes for what?

Comment (Cheney-W) [reply ports/ffmpeg/portfile.cmake:75]: I'm going to verify.

Comment (Cheney-W) [reply ports/ffmpeg/portfile.cmake]: After removing all download requirements under windows arm, `ffmpeg:arm-uwp` and `ffmpeg:arm64-windows` can still be installed successfully,  including all features.
I think they should be no longer needed now.

Comment (JackBoosY) [reply ports/ffmpeg/portfile.cmake]: I think this need @ras0219-msft to double confirm.

**Comment by LilyWangLL** [comment]
Ping @Cheney-W for response, could you please resolve the conflicts?

**Review by LilyWangLL — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**